README.md

```diff
@@ -1,3 +1,115 @@
 # libwhereami
 
-libwhereami detects virtualization environments.
+libwhereami is a C++11 library that detects virtualization environments from inside virtual machines.
+
+## Supported platforms
+
+- Linux
+- Windows
+- Mac OS
+- Solaris
+- AIX
+
+## Detected environments and reported metadata
+
+Not all metadata is consistently available on all platforms.
+
+- Docker
+  - id : string
+- Hyper-V
+- KVM
+  - openstack : bool
+  - google : bool
+- LXC
+  - name : string
+- OpenVZ
+  - id : int
+  - host : bool
+  - cloudlinux : bool
+- systemd-nspawn
+  - name : string
+- LDom
+  - chassis_serial : string
+  - control_domain : string
+  - domain_name : string
+  - domain_uuid : string
+  - role_control : bool
+  - role_io : bool
+  - role_root : bool
+  - role_service : bool
+- LPAR
+  - partition_name : string
+  - partition_number : int
+- VirtualBox
+  - version : string (requires root)
+  - revision : string (requires root)
+- VMware
+  - version : string (requires root)
+- WPAR
+  - key : int
+  - configured_id : int
+- Xen
+  - context : string ("hvm" or "pv")
+  - privileged : bool
+- Zone
+  - brand : string
+  - id : int
+  - ip_type : string
+  - name : string
+  - uuid : string
+
+## Example usage
+
+```cpp
```

Comment:
Doxygen appears to be falling over trying to do something with this block :(

Reply:
It seems like we could use the Doxyfile to disable checking for this (it actually explicitly includes README.md), but the Doxyfile is included in .gitignore... where could I change how the Doxyfile is generated?

Reply:
`Doxyfile.in` seems like the best place - we just do some simple variable substitution on that file, AFAICT

Reply:
D'oh, of course - updated, we'll see how this goes